From 201cba640f690cbc9ce6b9bd078283339daff201 Mon Sep 17 00:00:00 2001
From: Soren Brinkmann <soren.brinkmann@xilinx.com>
Date: Thu, 29 Sep 2016 10:32:07 -0700
Subject: [PATCH 0938/1566] ARM64: zynqmp: PM: Migrate to new shutdown
 interface

This patch comes from:
  https://github.com/Xilinx/linux-xlnx.git

Adjust to the revised firmware interface.

Signed-off-by: Soren Brinkmann <soren.brinkmann@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
(cherry picked from commit 0792aaca0a14816994a278174115bc7ba8815574)
---
 drivers/soc/xilinx/zynqmp/pm.c       |    6 +++---
 include/linux/soc/xilinx/zynqmp/pm.h |    2 +-
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/soc/xilinx/zynqmp/pm.c b/drivers/soc/xilinx/zynqmp/pm.c
index 67f0507..c532eae 100644
--- a/drivers/soc/xilinx/zynqmp/pm.c
+++ b/drivers/soc/xilinx/zynqmp/pm.c
@@ -361,9 +361,9 @@ EXPORT_SYMBOL_GPL(zynqmp_pm_set_wakeup_source);
  *
  * Return:	Returns status, either success or error+reason
  */
-int zynqmp_pm_system_shutdown(const u32 restart)
+int zynqmp_pm_system_shutdown(const u32 type, const u32 subtype)
 {
-	return invoke_pm_fn(SYSTEM_SHUTDOWN, restart, 0, 0, 0, NULL);
+	return invoke_pm_fn(SYSTEM_SHUTDOWN, type, subtype, 0, 0, NULL);
 }
 EXPORT_SYMBOL_GPL(zynqmp_pm_system_shutdown);
 
@@ -794,7 +794,7 @@ static ssize_t zynqmp_pm_debugfs_api_write(struct file *file,
 					pm_api_arg[1], pm_api_arg[2]);
 		break;
 	case SYSTEM_SHUTDOWN:
-		ret = zynqmp_pm_system_shutdown(pm_api_arg[0]);
+		ret = zynqmp_pm_system_shutdown(pm_api_arg[0], pm_api_arg[1]);
 		break;
 	case REQUEST_NODE:
 		ret = zynqmp_pm_request_node(pm_api_arg[0],
diff --git a/include/linux/soc/xilinx/zynqmp/pm.h b/include/linux/soc/xilinx/zynqmp/pm.h
index 891a896..394dc8f 100644
--- a/include/linux/soc/xilinx/zynqmp/pm.h
+++ b/include/linux/soc/xilinx/zynqmp/pm.h
@@ -209,7 +209,7 @@ int zynqmp_pm_request_wakeup(const u32 node,
 int zynqmp_pm_set_wakeup_source(const u32 target,
 					const u32 wakeup_node,
 					const u32 enable);
-int zynqmp_pm_system_shutdown(const u32 restart);
+int zynqmp_pm_system_shutdown(const u32 type, const u32 subtype);
 
 /* API for suspending of RPU */
 int zynqmp_pm_force_powerdown(const u32 target,
-- 
1.7.5.4

